internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c

```diff
@@ -0,0 +1,100 @@
+#include <vmlinux.h>
+
+#include <asm-generic/errno.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_helpers.h>
+
+#define MAX_ENTRIES 8 * 1024
+#define MAX_SYSCALLS 1024
+#define MAX_COMM_LEN 64
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, MAX_ENTRIES);
+    __type(key, u64);                            // namespace
+    __type(value, unsigned char[MAX_SYSCALLS]);  // syscall IDs
+} syscalls SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, MAX_ENTRIES);
+    __type(key, u32);    // PID
+    __type(value, u64);  // namespace
+} pids SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, MAX_ENTRIES);
+    __type(key, u32);                   // PID
+    __type(value, char[MAX_COMM_LEN]);  // command name
+} comms SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, 1 << 24);
+} events SEC(".maps");
+
+static __always_inline void * lookup_or_init(void * map, const u64 key)
+{
+    void * const value = bpf_map_lookup_elem(map, &key);
+    if (value) {
+        return value;
+    }
+
+    static const char init[MAX_SYSCALLS];
+    int err = bpf_map_update_elem(map, &key, &init, BPF_NOEXIST);
+    if (err && err != -EEXIST) {
+        return NULL;
+    }
+
+    return bpf_map_lookup_elem(map, &key);
+}
+
+SEC("tracepoint/raw_syscalls/sys_exit")
+int sys_exit(struct trace_event_raw_sys_exit * args)
+{
+    // Sanity check
+    u32 id = args->id;
+    if (id < 0 || id >= MAX_SYSCALLS) {
```

Comment:
Is this a sequential id for the `sys_exit` that starts from 0 every time we start a new BPF tracing per PID or is that kept throughout BPF tracing sessions?

Reply:
I'm not sure if I get that correctly. `id` is the syscall id, which we store in the `syscalls` map as value. :)